net/Socket.cpp

```diff
@@ -1395,6 +1407,60 @@ std::ostream& StreamSocket::stream(std::ostream& os) const
     return os << "]";
 }
 
+bool StreamSocket::checkRemoval(std::chrono::steady_clock::time_point now)
+{
+    if (!isIPType()) // forced removal on IPv[46] network connections only
+    {
+        return false;
+    }
+    const auto durTotal =
+        std::chrono::duration_cast<std::chrono::milliseconds>(now - getCreationTime());
+    const auto durLast =
+        std::chrono::duration_cast<std::chrono::milliseconds>(now - getLastSeenTime());
+    const double bytesPerSecIn = durTotal.count() > 0 ? (double)bytesRcvd() / ((double)durTotal.count() / 1000.0) : 0.0;
+    const bool c1 = now < getCreationTime();
+    const bool c2 = _pollTimeout > std::chrono::microseconds::zero() &&
```

Comment:
This is very complex. Are there any tests ?
the _pollTimeout I would expect to sometimes be very small - depending on timers - so say it is 1 microsecond, but normally say 5 seconds or somesuch; and the durLast - is the gap to the last seen time can easily be longer than that - so - I would expect c2 to regularly be true, and depend if someone setup a short timeout in the loop - right ?

Can we simplify these - and give them useful symbolic names to aid comprehension ? =) c1 -> c4 don't float my boat :-)

Reply:
- c2 (now cIDLE) denotes: `TO Criteria: Violate maximum idle (_pollTimeout default 64s)`
- renamed the criteria names as well as their logging description